src/server/snap.rs

```diff
@@ -239,6 +239,9 @@ impl<R: RaftStoreRouter + 'static> Runnable<Task> for Runner<R> {
                 SNAP_TASK_COUNTER.with_label_values(&["write"]).inc();
                 match self.files.entry(token) {
                     Entry::Occupied(mut e) => {
+                        let mut copy = self.snap_mgr.clone();
```

Comment:
seem you don't need clone here.

Reply:
In line 254 `self.snap_mgr.deregister` here snap_mgr is immutable, so if i just call `self.snap_mgr.get_limiter()` it will cause a compile error since here snap_mgr is mutable.

Reply:
please change your `get_limiter` above.